ocaml/xapi/xapi_observer.ml

```diff
@@ -381,7 +392,10 @@ let set_attributes ~__context ~self ~value =
         Forwarder.set_attributes ~__context ~uuid
           ~attributes:(default_attributes @ value)
       )
-      (observed_components_of (Db.Observer.get_components ~__context ~self))
+      (Db.Observer.get_components ~__context ~self
+      |> List.map Component.of_string
+      |> observed_components_of
+      )
```

Comment:
Maybe we could let 
```OCaml
   Db.Observer.get_components ~__context ~self
   |> List.map Component.of_string
   |> observed_components_of
```` 
to be a function here, it appears three times in this place.